engine/src/test/java/com/sandboni/core/engine/sta/SeloniJsonConnectorTest.java

```diff
@@ -0,0 +1,108 @@
+import com.sandboni.core.engine.sta.graph.Link;
+import com.sandboni.core.engine.sta.graph.LinkFactory;
+import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.CucumberVertex;
+import com.sandboni.core.engine.sta.graph.vertex.TestVertex;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+import com.sandboni.core.scm.scope.ChangeScopeImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+public class SeloniJsonConnectorTest {
+
+    private String sampleVerb =  HttpConsts.getHttpVerb().stream().findFirst().get();
+
+    private Connector httpTemplateConnector;
+
+    private Context setupContext(String calleeAction) {
+        Context context = new Context(new String[0], new String[0], "", new ChangeScopeImpl());
+
+        if (Objects.nonNull(calleeAction)) {
+            context.addLink(LinkFactory.createInstance(
+                    context.getApplicationId(),
+                    new Vertex.Builder(sampleVerb + " " + HttpConsts.HTTP_LOCALHOST, calleeAction)
+                            .markSpecial()
+                            .build(),
+                    new Vertex.Builder("callee", "action").build() ,
+                    LinkType.HTTP_HANLDER));
+        }
+        return context;
+    }
+
+    @Before
+    public void setup(){
+        System.setProperty(SystemProperties.SELONI_FILEPATH.getName(), "./src/test/resources/Seloni.json" );
+        httpTemplateConnector = new HttpTemplateConnector();
+    }
+
+    @Test
+    public void testMultipleLinksCucumber(){
+        Context context = setupContext("/rest/unprocess/count/NEW");
+        httpTemplateConnector.connect(context);
+
+        assertEquals(7, context.getLinks().count());
+
+        Optional<Link> result = context.getLinks().filter(l -> l.getCaller().getAction().equals("test-name") && l.getCallee().getAction().equals("/rest/unprocess/count/NEW") && l.getLinkType() == LinkType.HTTP_MAP_SELONI).findFirst();
+        Assert.assertTrue(result.isPresent());
+    }
+
+    @Test
+    public void testLinkCucumber(){
+        Context context = setupContext("/rest/unprocess/count/ABC");
+        httpTemplateConnector.connect(context);
+
+        assertEquals(6, context.getLinks().count());
+
+        Optional<Link> result = context.getLinks().filter(l -> l.getCaller().getAction().equals("test-name3") && l.getCallee().getAction().equals("/rest/unprocess/count/ABC") && l.getLinkType() == LinkType.HTTP_MAP_SELONI).findFirst();
+        Assert.assertTrue(result.isPresent());
+
+        Link link = result.get();
+
+        assertTrue(link.getCaller() instanceof  CucumberVertex);
+
+        assertEquals(9, ((CucumberVertex)link.getCaller()).getScenarioLine());
+        assertEquals("I:\\code\\seloni-demo-test\\className2.class", ((CucumberVertex)link.getCaller()).getFeaturePath());
```

Comment:
change this path to make it work in windows and linux

Reply:
it's just comparing strings. this is the same featurePath value that exists on the test json file. on real life it contains the right path